lib/public/IDBConnection.php

```
@@ -339,4 +344,13 @@ public function createSchema(): Schema;
 	 * @since 13.0.0
 	 */
 	public function migrateToSchema(Schema $toSchema): void;
+
+	/**
+	 * Returns the database provider name
+	 *
+	 * @throws Exception
```

Avoid throwing if possible

---

well if the db isn't there then the exception can't be handled any way so I won't catch it and let it bubble up.